main.tf

```diff
@@ -188,8 +192,10 @@ locals {
 }
 
 resource "aws_cloudfront_distribution" "default" {
+  count = module.this.enabled ? 1 : 0
+
   #bridgecrew:skip=BC_AWS_LOGGING_20:Skipping `CloudFront Access Logging` check until bridgecrew will support dynamic blocks (https://github.com/bridgecrewio/checkov/issues/776).
-  enabled             = module.this.enabled
+  enabled             = var.distribution_enabled
```

Comment:
I think this should be `count  = module.this.enabled && var.distribution_enabled` otherwise you could have a bucket without a cloudfront distribution and that is not the purpose of this module.

Reply:
The `enabled` you referred to is whether the CF Distribution needs to be enabled or not. The count is for that resource is `count = module.this.enabled ? 1 : 0` , which prevents the CF distribution from being created when the cf-s3-cdn module is set to `enabled = false`.

If we change the `count` for this module to the one you mentioned, we cannot toggle the status of the CF Distribution itself (on AWS side).

Reply:
This module primarily created a CF distribution and a public s3 bucket ( plus some other stuff like logs, permissions etc) so if we allow for the CF distribution to be disabled, then we are creating a public bucket basically and that is not what the module is meant for, I think

Reply:
I see ok, sorry it is very rare to have an `enabled` flag on a resource.